### ISSUE TITLE
Implement LapComparator for Actual vs Ideal Analysis

### ISSUE THREAD

**Issue by yannicktuerk** [yannicktuerk/F1-Lap-Bot#45]
## Summary
Compare actual lap telemetry against ideal lap to identify time loss segments and classify driving errors. Core logic for Mathe-Coach feedback generation.

## Technical Details
Create `LapComparator` domain service in `src/domain/services/lap_comparator.py`:

### Comparison Algorithm:
```python
@dataclass
class ComparisonSegment:
    """Results for one track segment."""
    segment_id: str
    distance_start: float
    distance_end: float
    time_loss: float           # seconds lost in this segment
    speed_delta_entry: float   # km/h difference at entry
    speed_delta_apex: float    # km/h difference at apex
    speed_delta_exit: float    # km/h difference at exit
    error_type: str            # "early_braking", "slow_corner", "late_throttle", etc.
    explanation: str

class LapComparator:
    def compare_laps(
        self,
        actual_lap: LapTrace,
        ideal_lap: IdealLap,
        track_profile: TrackProfile,
        segments: List[TrackSegment]
    ) -> List[ComparisonSegment]:
        """
        Compare actual vs ideal lap, segment by segment.
        
        Returns prioritized list of comparison results.
        """
        results = []
        
        for segment in segments:
            # Extract samples in this segment
            actual_samples = [
                s for s in actual_lap.samples
                if segment.distance_start <= s.lap_distance <= segment.distance_end
            ]
            
            if not actual_samples:
                continue
            
            # Compute metrics
            entry_speed_actual = actual_samples[0].speed
            entry_speed_ideal = ideal_lap.get_speed_at(segment.distance_start)
            
            # Find apex (min speed in high-curvature region)
            apex_idx = np.argmax([track_profile.get_curvature_at(s.lap_distance) for s in actual_samples])
            apex_speed_actual = actual_samples[apex_idx].speed
            apex_distance = actual_samples[apex_idx].lap_distance
            apex_speed_ideal = ideal_lap.get_speed_at(apex_distance)
            
            exit_speed_actual = actual_samples[-1].speed
            exit_speed_ideal = ideal_lap.get_speed_at(segment.distance_end)
            
            # Compute time loss
            time_loss = sum(
                ideal_lap.get_time_loss_at(s.lap_distance, s.speed) * 10.0  # ds
                for s in actual_samples
            )
            
            # Classify error
            error_type, explanation = self._classify_error(
                actual_samples, ideal_lap, segment
            )
            
            results.append(ComparisonSegment(
                segment_id=segment.id,
                distance_start=segment.distance_start,
                distance_end=segment.distance_end,
                time_loss=time_loss,
                speed_delta_entry=(entry_speed_actual - entry_speed_ideal) * 3.6,  # m/s to km/h
                speed_delta_apex=(apex_speed_actual - apex_speed_ideal) * 3.6,
                speed_delta_exit=(exit_speed_actual - exit_speed_ideal) * 3.6,
                error_type=error_type,
                explanation=explanation
            ))
        
        # Sort by time_loss descending
        results.sort(key=lambda x: x.time_loss, reverse=True)
        return results
```

### Error Classification:
- **"early_braking"**: brake_onset before ideal
- **"late_braking"**: brake_end after ideal
- **"slow_corner"**: apex_speed < ideal in high-κ section
- **"late_throttle"**: throttle_full_on after ideal on exit
- **"line_error"**: lateral offset too large

## Checklist
- [ ] Implement telemetry alignment by lap_distance
- [ ] Calculate speed and time deltas per segment
- [ ] Classify driving errors based on telemetry patterns
- [ ] Prioritize feedback items by time loss magnitude
- [ ] Create ComparisonSegment dataclass
- [ ] Write unit tests with synthetic actual/ideal laps
- [ ] Test edge cases (perfect lap, very different lap)

## Acceptance Criteria
- LapComparator accurately identifies time loss areas
- Error classification makes sense for typical mistakes
- Prioritization surfaces most impactful improvements first

## References
- Spec: `warp_prompt_f1_lap_bot.md` Section 5.5 (Mathe-Coach Feedback)
